src/unix/fs.c

```diff
@@ -231,7 +231,68 @@ static ssize_t uv__fs_futime(uv_fs_t* req) {
 
 
 static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
+#if defined(__MVS__)
+  /* There is no mkdtemp. So instead use mktemp to generate a
+     temporary file. Then delete that file and use the name
+     to create a temporary directory.
+  */
+  char *path = (char*) req->path;
+  static const char* tempchars =
+    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  static const size_t num_chars = 62;
+  static const size_t num_x = 6;
+  char *ep, *cp;
+  unsigned int tries, i;
+  size_t len;
+  uint64_t v;
+  FILE *finput;
+  int retval;
+  int saved_errno;
+
+  len = strlen(path);
+  ep = path + len;
+  if (len < num_x || strncmp(ep - num_x, "XXXXXX", num_x)) {
+    errno = EINVAL;
+    return -1;
+  }
+
+  finput = fopen("/dev/urandom", "r");
```

Comment:
edit: maybe in `/os390-syscalls.c`? It would keep the os390 compat layer together, and minimize os390 code in the more generic files.